### DISCUSSION TITLE
[Support] 使用pm2启动时，pm2和midway日志重复

### DISCUSSION THREAD

**Discussion by minefujiko9293** [midwayjs/midway#3716] [Q&A] [answered]
### Describe the problem（描述问题）

使用pm2启动后，pm2也会有日志记录，重复日志浪费空间。
哪到底是停用pm2的日志还是mw的日志好呢。

### Midway Versions（Midway 版本）

>> Start to check your midway component version...

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(3.15.0)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.15.0)
✓ @midwayjs/bull(3.15.3)
✓ @midwayjs/bull-board(3.15.3)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.15.0)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(3.15.2)
✓ @midwayjs/decorator(3.15.0)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(3.15.2)
✓ @midwayjs/info(3.15.2)
✓ @midwayjs/jwt(3.15.2)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.15.2)
✓ @midwayjs/mongoose(3.15.3)
✓ @midwayjs/mqtt(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(3.15.2)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(3.15.2)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(3.15.3)
✓ @midwayjs/typeorm(not installed)
✓ @midwayjs/upload(3.15.2)
✓ @midwayjs/validate(3.15.2)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.15.2)
✓ @midwayjs/ws(not installed)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by czy88840616** [comment]
pm2记录的是 stdout/stderr，midway 记录的是 logger 实例打印的，不一样的。

**Reply by minefujiko9293** [reply]
```

import { Catch } from '@midwayjs/core';
import { Context } from '@midwayjs/koa';

@Catch()
export class DefaultErrorFilter {
  async catch(err: any, ctx: Context) {
    const hint = ctx.startTime;

    // 所有的未分类错误会到这里
    console.warn('/*-[DefaultErrorFilter]-*/ catch:');
    ctx.logger.error(`[${hint}]`, err);
    console.warn('/*-[DefaultErrorFilter]-*/');

    const errcode = err.status ?? 500;
    const errmsg = err.toString() || err.message || 'Internal Server Error';

    return {
      errcode,
      errmsg: `(Hint:${hint}) ` + errmsg,
    };
  }
}

```

好像这个中间件记录的错误信息，mw和pm2都重复记录了ctx.logger.error(`[${hint}]`, err);

是不是midway的logger的transports.console也包含了控制台输出

**Reply by czy88840616** [reply, marked as answer]
这么看是的，我看了下 logger 好像在线上没禁用，我调整下。
